dotcom-rendering/src/components/LatestLinks.importable.tsx

```diff
@@ -0,0 +1,199 @@
+	padding: 0 ${space[1]}px;
+`;
+
+const li = css`
+	${textSans.xsmall()}
+	overflow: hidden;
+	flex-grow: 1;
+	height: calc(4 * 1.35em + 2 * ${space[1]}px);
+`;
+
+const dividerStyles = css`
+	border-top: 1px solid currentColor;
+	border-left: 1px solid currentColor;
+`;
+
+const bold = css`
+	${textSans.xsmall({ fontWeight: 'bold' })};
+
+	:before {
+		content: '';
+		height: 0.75em;
+		width: 0.75em;
+		margin-right: ${space[1]}px;
+		display: inline-block;
+		background-color: currentColor;
+		border-radius: 100%;
+	}
+`;
+
+const transparent = css`
+	color: transparent;
+`;
+
+const Time = ({
+	epoch,
+	colour,
+}: {
+	epoch: number;
+	colour: SerializedStyles;
+}) => (
+	<>
+		{' '}
+		<strong css={[bold, colour]}>
+			<Hide until="tablet">{timeAgo(epoch, true)}</Hide>
+			<Hide from="tablet">{timeAgo(epoch, false)}</Hide>
+		</strong>
+		<br />
+	</>
+);
+
+/**
+ * # Latest Links
+ *
+ * Display the last three blocks from a Liveblog when the fronts
+ * tool has enabled showing live updates.
+ *
+ * ## Why does this need to be an Island?
+ *
+ * We make an API call to retrieve the latest blocks,
+ * constantly updating when new data comes in.
+ *
+ * ---
+ *
+ * (No visual story exists)
+ */
+export const LatestLinks = ({
+	id,
+	format,
+	direction,
+	isDynamo,
+	containerPalette,
+}: Props) => {
+	const { data } = useApi<Blocks>(
+		`https://api.nextgen.guardianapps.co.uk${id}.json?rendered=false`,
+		{
+			compare: () => false, // even if the data is the same, we still want to re-render so that the time updates
+			refreshInterval: 9_600, // this number was carefully
+		},
+	);
+
+	const { text } = decidePalette(format, containerPalette);
+	const kickerColour = isDynamo ? text.dynamoKicker : text.cardKicker;
+
+	const colour = css`
+		color: ${kickerColour};
+	`;
+
+	const dividerColour = css`
+		color: ${containerPalette
+			? decideContainerOverrides(containerPalette).border?.container
+			: palette.neutral[86]};
+	`;
+
+	return (
+		<ul
+			css={[
+				style,
+				revealStyles,
+				isDynamo || direction === 'horizontal' ? horizontal : vertical,
+				css`
+					color: ${text.cardHeadline};
+				`,
+			]}
+		>
+			{data && data.blocks.length >= 3 ? (
+				data.blocks.slice(0, 3).map((block, index) => (
+					<>
+						{index > 0 && (
+							<li
+								key={block.id + ' : divider'}
+								css={[dividerStyles, dividerColour]}
+							></li>
+						)}
+						<li key={block.id} css={li} className={'reveal'}>
+							<WithLink
+								linkTo={`${id}?page=with:block-${block.id}#block-${block.id}`}
+								isDynamo={isDynamo}
+							>
+								<Time
+									epoch={block.publishedDateTime}
+									colour={colour}
+								/>
+								<span className="show-underline">
+									{
+										block.body
+											.slice(0, 75) // take the first couple of characters
+											.split(' ') // split by words
+											.slice(0, -1) // drop the last word
+											.join(' ') + '…' // join with spaces and add an ellipsis
```

Comment:
could we achieve this with https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow? and if not may be worth commenting as to why so someone doesn't come along and replace this code?

Reply:
I’ve commented in the PR description, but `text-overflow` only works for single lines.

Reply:
👍